.github/workflows/clang-format.yaml

```diff
@@ -0,0 +1,52 @@
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+#
+# SPDX-FileCopyrightText: Copyright 2022 Micron Technology, Inc.
+
+name: ClangFormat
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
+    branches:
+      - master
+      - "v[0-9]+.[0-9]+"
+    paths:
+      - "**.c"
+      - "**.h"
+      - "**.h.in"
+      - "!subprojects/*"
+      - .github/workflows/clang-format.yaml
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref }}
+  cancel-in-progress: true
+
+permissions: {}
+
+jobs:
+  clang-format:
+    runs-on: ubuntu-latest
+    container:
+      image: ghcr.io/hse-project/ci-images/fedora-37:${{ github.base_ref }}
+
+    steps:
+      - name: Checkout HSE
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 2
+
+      - name: Get clang-format
+        run: |
+          dnf install clang-tools-extra
+
+      - name: Run clang-format
+        run: |
+          ./scripts/dev/clang-format.sh -c "$GITHUB_WORKSPACE"
```

Comment:
Is there any way for this tooling to know which files have changed in the PR?  If so, why not run `scripts/dev/format.sh -c <files>` ?  It would be much simpler and it produces diagnostic output.

Reply:
Not easily. I don't think it will make much difference now that the repo has been properly formatted.